src/static/riot/analytics/analytics.tag

```diff
@@ -0,0 +1,520 @@
+
+            general_calendar_options = {
+                type: 'date',
+                // Sets the format of the placeholder date string to YYYY-MM-DD
+                formatter: {
+                   date: function (date, settings) {
+                       if (!date) return '';
+                       var day = date.getDate();
+                       var month = date.getMonth() + 1;
+                       var year = date.getFullYear();
+                       return year + '-' + month + '-' + day;
+                   }
+                },
+            }
+
+            start_specific_options = {
+                endCalendar: $(self.refs.end_calendar),
+                onChange: function(date, text) {
+
+                    let year = date.getFullYear()
+                    let month = date.getMonth()
+                    if(month === 0) {
+                        start_date = {
+                            year: year - 1,
+                            month: 12,
+                            day: 1,
+                        }
+                    } else {
+                        start_date = {
+                            year: year,
+                            month: month,
+                            day: 1,
+                        }
+                    }
+                    self.update_analytics(start_date, end_date, self.time_unit)
+                },
+            }
+
+            end_specific_options = {
+                startCalendar: $(self.refs.start_calendar),
+                onChange: function(date, text) {
+
+                    let year = date.getFullYear()
+                    let month = date.getMonth() + 1
+                    let day = date.getDate()
+                    end_date = {
+                        year: year,
+                        month: month,
+                        day: day,
+                    }
+
+                    self.update_analytics(start_date, end_date, self.time_unit)
+                },
+            }
+
+            start_calendar_options = _.assign({}, general_calendar_options, start_specific_options)
+            end_calendar_options = _.assign({}, general_calendar_options, end_specific_options)
+
+            $(self.refs.start_calendar).calendar(start_calendar_options);
+            $(self.refs.end_calendar).calendar(end_calendar_options);
+
+            /*---------------------------------------------------------------------
+             Chart Setup
+            ---------------------------------------------------------------------*/
+
+            var ctx_c = document.getElementById('competition_chart').getContext('2d');
+            self.competitionsChart = new Chart(ctx_c, create_chart_config('# of Competitions'));
+            var ctx_s = document.getElementById('submission_chart').getContext('2d');
+            self.submissionsChart = new Chart(ctx_s, create_chart_config('# of Submissions'));
+            var ctx_u = document.getElementById('user_chart').getContext('2d');
```

Comment:
Can probably do all three of these w/ `self.refs.` instead of `getElementById`s

Reply:
done.